src/pages/admin/donate/components/generic-details/GenericDetails.test.tsx

```
@@ -0,0 +1,203 @@
+
+        const newItems = onChangeItems.mock.calls[0][0];
+
+        rerender(<GenericDetails {...defaultProps} items={newItems} onChangeItems={onChangeItems} />);
+
+        await waitFor(() => {
+            expect(screen.getByText('New Item')).toBeInTheDocument();
+        });
+    });
+
+    it('renders children function for each item', () => {
+        render(<GenericDetails {...defaultProps} children={({ formState }) => <span>{formState.name}</span>} />);
```

Biome (2.1.2)

[error] Avoid passing children using a prop

The canonical way to pass children in React is to use JSX elements

(lint/correctness/noChildrenProp)